active-rfcs/0000-component-tag.md

```diff
@@ -0,0 +1,242 @@
+
+### 2. Remove component script support entirely
+
+- Remove `<script hoist>` support and do not replace it with anything.
+- If you want bundled script behavior, create a client-side component (React, Vue, etc).
+
+In this case, the rest of this proposal would apply to `<style>` only.
+
+I think we can reuse enough logic from how `client:only` works that this may not cost us much in terms of implemention. In which case, I think it makes sense to keep. Still TBD if this is true though, will need a prototype to confirm.
+
+## Alternatives - Naming
+
+Note: While naming is important, please make sure that you also give feedback on the **behavior** that is being proposed. The behavior of this proposal is more complex and by far the most important piece to finalize.
+
+### 1. Alternative tag name
+
+```
+<style @component>
+// or:
+<style @page>
+<style @astro>
+<style @bundle>
+```
+
+- Main reason for this is that `@component` may be confusing to use on a page. I believe that this is not a big issue, but a different name could reduce this confusion.
+
+### 2. Skip the new syntax, just use a directive
+
+```
+<style astro:component>
+// or:
+<style is:component>
+<style is:bundled>
+<style astro:build>
+```
+
+- Pro: Already supported by the compiler
+- Pro: No new rules need to be added to our language
+- Pro: Would not conflict with [Alpine.js event handlers](https://alpinejs.dev/essentials/events#listening-for-simple-events) (see "Unresolved questions" below)
+- Con: This behavior is much more complex than our other directives, so is directive really a good fit? For example, this would be the first directive to state that no other attributes can exist on a `script/style` tag because it is effectively virtual and must run as expected in the final bundle.
+
+### 3. Use a special "virtual" component syntax
+
+```
+<astro:style>    // or <component:style>
+<astro:script> // or <component:script>
+```
+
+- Pro: Precedence, similar to `<svelte:head>`, etc.
+- Pro: Even more unique, which makes it easier to define our own behavior without confusing for normal `<style>` and `<script>` expectations
+- Pro: Similarly easy to statically analyze
+- Con: Not currently supported in our compiler
+- Con: Looks like a directive, which is syntactically confusing for directive attributes
+- Con: Both start and end tags would need to be updated `<astro:style></astro:style>`
+
+### 4. Use a special Astro component
+
+```astro
+---
+import {Style, Script} from 'astro/components';
+---
+<Style>...</Style>
+<Script>...</Script>
+```
+
+- Pro: No user has an expectation about how this would work. That means less conflict with how `<style>` and `<script>` are expected to work.
+- Con: Requires extra import boilerplate in every component. We could make these globally available to mitigate, but we've never done that before.
```

Comment:
We have one global component, `Fragment`. I think the primary problem with a Component is that it can be renamed (`const MyStyle = Style`) and its hard to warn against doing that.